src/Test/Utilities/CLRHelpers.cs

```diff
@@ -64,12 +65,14 @@ public static string[] PeVerify(string filePath)
             return PeVerify(File.ReadAllBytes(filePath), AppDomain.CurrentDomain.Id, filePath);
         }
 
-        private static readonly object Guard = new object();
+        private static readonly Mutex s_guard = new Mutex(false, new Guid().ToString());
```

Comment:
This creates system-wide named mutex, right? I don't think we want be taking system-wide locks.